controllers/mover/syncthing/syncthing_test.go

```diff
@@ -474,7 +477,143 @@ var _ = Describe("When an RS specifies Syncthing", func() {
 		})
 
 		Context("Syncthing API is being used properly", func() {
-			// TODO: test syncthing API
+			When("Syncthing server does not exist", func() {
+				It("Doesn't synchronize", func() {
+					res, err := mover.Synchronize(ctx)
+					Expect(err).ToNot(BeNil())
+					Expect(res).To(Equal(cMover.InProgress()))
+
+					err = mover.syncthing.FetchSyncthingConfig()
+					Expect(err).ToNot(BeNil())
+					Expect(err.Error()).To(ContainSubstring("Get"))
+
+					err = mover.syncthing.FetchSyncthingSystemStatus()
+					Expect(err).ToNot(BeNil())
+					Expect(err.Error()).To(ContainSubstring("Get"))
+
+					err = mover.syncthing.FetchConnectedStatus()
+					Expect(err).ToNot(BeNil())
+					Expect(err.Error()).To(ContainSubstring("Get"))
+
+					apiKeys := &corev1.Secret{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "volsync-" + mover.owner.GetName(),
+							Namespace: ns.Name,
+						},
+						Data: map[string][]byte{
+							"apikey":   []byte("my-secret-apikey-do-not-steal"),
+							"username": []byte("gcostanza"),
+							"password": []byte("bosco"),
+						},
+					}
+
+					err = mover.ensureIsConfigured(apiKeys)
+					Expect(err).ToNot(BeNil())
+					Expect(err.Error()).To(ContainSubstring("no such host"))
+					Expect(err.Error()).To(ContainSubstring("Get"))
+
+					service := &corev1.Service{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      mover.getAPIServiceName(),
+							Namespace: mover.owner.GetNamespace(),
+						},
+					}
+					err = mover.ensureStatusIsUpdated(service)
+					Expect(err).ToNot(BeNil())
+					Expect(err.Error()).To(ContainSubstring("no such host"))
+					Expect(err.Error()).To(ContainSubstring("Get"))
+
+					_, err = mover.syncthing.jsonRequest("/rest/config", "GET", nil)
+					Expect(err).ToNot(BeNil())
+					Expect(err.Error()).To(ContainSubstring("no such host"))
+					Expect(err.Error()).To(ContainSubstring("Get"))
+				})
+			})
+
+			When("Syncthing server exists", func() {
```

Comment:
Great work so far! The only thing missing in this section now are tests for the `mover.ensureStatusIsUpdated` and `mover.ensureIsConfigured` methods. Once we have that, this PR should be good to merge.